app.py

```diff
@@ -528,6 +569,40 @@ def _filter_carpools_by_date(leave_time, return_time):
     return pools
 
 
+def _filter_carpools_by_location(from_latitude,
+                                 from_longitude,
+                                 to_latitude,
+                                 to_longitude):
+
+    pools = Carpool.query.filter(
+        Carpool.from_latitude == from_latitude,
```

Comment:
I'm not sure I follow. There are no operations being performed on floats (avoiding the pitfalls of float arithmetic), so those values should stay the same (we can even make them string fields and only convert them in on the front-end). Right now, the user simply clicks on a marker and is presented with a series of other markers to click on. So long as no one decides to actually add or multiply lat/lng floats, the values should just match. Unless I'm missing an obvious point about lat/lng attributes changing for addresses over time! However, even in this case, we're only reading the values from google maps api once (when the driver enters their address). We then populate the map and do our filtering from our own stored values.

If you're implying that the user should be able to search in a better way than by clicking on a marker, I totally agree, but that would require some design work.

Apologies if I'm missing something utterly obvious.